manim/animation/animation.py

```diff
@@ -60,7 +60,7 @@ def __init__(
                 )
             )
 
-    def _typecheck_input(self, mobject: Mobject) -> None:
+    def _typecheck_input(self, mobject: Optional[Mobject]) -> None:
```

Comment:
It would make more sense to have it as a keyword argument with default value as `None`
```suggestion
    def _typecheck_input(self, mobject: Optional[Mobject] = None) -> None:
```

Reply:
This I don't understand at all!! 
Why would you supply default arguments to a function written only to typecheck other arguments?

but more generally, I want to get rid of each and every _typecheck_input we have. The idea is to first fix all mypy errors, and then add a decorator that typechecks arguments without the need for us to do it manually.

Reply:
Think of "None" as acting like a special type of empty mobject as opposed to the absence of a mobject.